### PR TITLE
AG-5670 Join duplicate bar-like series code into AbstractBarSeries

### PR DESCRIPTION
**Pull request by olegat**
https://ag-grid.atlassian.net/browse/AG-5670

Cut down duplicate code in the follow series classes:
- `BarSeries`
- `BoxPlotSeries`
- `BullerSeries`
- `RangeBar`
- `WaterfallSeries`

This merges common features into a new `AbstractBarSeries`. Right now, this class just contains a `direction` property and methods to help with category/number axes management.

### CODE REVIEW

File: packages/ag-charts-enterprise/src/series/box-plot/blotPlotUtil.ts
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/series/box-plot/blotPlotUtil.ts:1]: Please undo this change, this was following he same pattern as other series for animation handling, and once add/update/remove/legend-toggle cases are added the module is likely to be much more complex.

**Review by alantreadway — COMMENTED** [review]
If you could address the one comment, this looks good - thanks for the clean-up.

As a heads-up, the intent is to move to composed series implementation longer-term, so I would avoid pulling more behaviour into these classes where possible (our general approach has been to break-out reusable functions for a bunch of cases).

File: packages/ag-charts-enterprise/src/series/box-plot/blotPlotUtil.ts
Comment (olegat) [reply packages/ag-charts-enterprise/src/series/box-plot/blotPlotUtil.ts:1]: Ah, OK. Thanks for providing context. I'll roll this change back.

**Comment by olegat** [comment]
> If you could address the one comment, this looks good - thanks for the clean-up.
> 
> As a heads-up, the intent is to move to composed series implementation longer-term, so I would avoid pulling more behaviour into these classes where possible (our general approach has been to break-out reusable functions for a bunch of cases).

Noted. I'll do that where applicable.
